utils/logger.py

```diff
@@ -0,0 +1,149 @@
+        headers = {
+            "Authorization": f"Bearer {self.token}",
+            "Content-Type": "application/json",
+        }
+        data = {"duration": 86400}  # Set to 24 hours
+
+        try:
+            response = requests.post(url, headers=headers, json=data)
+            response.raise_for_status()
+            impersonate_data = response.json()
+            self.impersonate_token = impersonate_data["token"]
+
+        except requests.exceptions.HTTPError as err:
+            raise Exception(f"Failed to impersonate: {err}")
+
+    def _collection_exists(self):
+        if not self.impersonate_token:
+            return False
+
+        url = f"{self.url}/api/collections/{self.collection_name}"
+        headers = {
+            "Authorization": f"Bearer {self.impersonate_token}",
+            "Content-Type": "application/json",
+        }
+        response = requests.get(url, headers=headers)
+        return response.status_code == 200
+
+    def _create_log_collection(self):
+        if not self.impersonate_token:
+            return
+
+        url = f"{self.url}/api/collections"
+        headers = {
+            "Authorization": f"Bearer {self.impersonate_token}",
+            "Content-Type": "application/json",
+        }
+
+        collection_data: dict[str, Any] = {
+            "name": self.collection_name,
+            "type": "base",
+            "fields": [
+                {"name": "PackageIdentifier", "type": "text"},
+                {"name": "DepositGroupIdentifier", "type": "text"},
+                {"name": "Level", "type": "text", "required": True},
+                {"name": "Message", "type": "text", "required": True},
+                {
+                    "name": "Timestamp",
+                    "type": "autodate",
+                    "onCreate": True,
+                    "onUpdate": False,
+                },
+            ],
+            "system": False,
+        }
+
+        response = requests.post(url, json=collection_data, headers=headers)
+
+        if not response.status_code == 200:
+            raise Exception(
+                f"Failed to save logs: {response.status_code}, {response.text}"
+            )
+
+    def _write_log(self, log_data: dict[str, Any]):
+        if not self.impersonate_token:
+            return
+
+        log_data["Timestamp"] = datetime.now().isoformat()
+        if not self._collection_exists():
+            self._create_log_collection()
+
+        url = f"{self.url}/api/collections/{self.collection_name}/records"
+        headers = {
+            "Authorization": f"Bearer {self.impersonate_token}",
+            "Content-Type": "application/json",
+        }
+
+        response = requests.post(url, json=log_data, headers=headers)
+
+        if not response.status_code == 200:
+            raise Exception(
+                f"Failed to save logs: {response.status_code}, {response.text}"
+            )
```

Comment:
Maybe create a `LoggerException` that you can raise instead of the base exception?